modules/canopennode/Kconfig

```diff
@@ -8,6 +8,7 @@ config ZEPHYR_CANOPENNODE_MODULE
 
 config CANOPENNODE
 	bool "CANopenNode support"
+	select CRC
```

Comment:
same here. `depends on` should be preferred over `select` as per https://docs.zephyrproject.org/latest/build/kconfig/tips.html#select-recommendations.

Reply:
why? where would CRC be enabled then? In the application? This is no different from `select BASE64` we already have and other libraries without any dependencies, which is also mentioned in the tips page above.

Reply:
> where would CRC be enabled then? In the application?

yes, and that is not necessarily a bad thing (that way users become aware of what they are really getting into -- e.g. what if it were not just `CRC` and something like `POSIX_API`?).

> This is no different from `select BASE64` we already have and other libraries without any dependencies

sure, until someone comes and adds `depends on` to these for whatever reason in the future.

no strong opinion since this file seems to have been already infected by the misuse of `select`, but they will likely need to be cleaned up at some point in the future.